src/video/wayland/SDL_waylandevents.c

```diff
@@ -661,6 +662,30 @@ keyboard_handle_keymap(void *data, struct wl_keyboard *keyboard,
         input->xkb.keymap = NULL;
         return;
     }
+
+    /*
+     * See https://blogs.s-osg.org/compose-key-support-weston/
+     * for further explanation on dead keys in Wayland.
+     */
```

Comment:
Looks like this link is pretty dead even on Wayback Machine, so we can probably replace this with the official documentation for dead keys.